### ISSUE TITLE
Url-Bar suggestions drawing behind expanded sidebar

### ISSUE THREAD

**Issue by klubklubla** [ranmaru22/firefox-vertical-tabs#138]
Hi there,

I am verry happy using this combination of userchrome and tab center from ranmaru. I am using it a long time now. Now I had set the expanding suggestion box of the url-bar to draw on top of the expanded side bar, but in a recent firefox update, this has been reversed.

I updated tab center reborn and userchrome css with the most recent stable version (mini), but unfortunately this did not solve my issue. I want the suggestion box of the url bar to always draw over the expanded side bar.
![Bildschirmfoto vom 2023-11-03 13-34-37](https://github.com/ranmaru22/firefox-vertical-tabs/assets/149794407/c2a3d95e-8728-4ba7-a038-f0c923f8ec4b)

I am sure this is fairly basic. Can you help me out? Thanks!
![Bildschirmfoto vom 2023-11-03 13-34-37](https://github.com/ranmaru22/firefox-vertical-tabs/assets/149794407/cbc5521c-7d7e-425f-b013-2825e8efd9cb)

**Comment by ranmaru22** [comment]
This is a really strange one which multiple people on different platforms seem to encounter. The problem is, I could never really figure out what the issue is.

I thought it's something to do with the sidebar's z-index, but even if I crank that up to some really high number, it works for me. The issue seems to have something to do with what happens to `.urlbarView` when you focus the input. Normally it's set to `display: none`, and then it changes to `block`, but that's not all that's happening. The behavior of this bit of the UI depends on some Firefox internals which user CSS can't touch, namely some custom XML props on `#urlbar`. When you focus it, it gains several custom props like `focus=true`, `open`, and `breakout-extend=true`, and those determine whether the popup part of the URL bar extends beyond its normal box.

Without it, even with a z-index close to maximum and forced display, it won't show. 

![image](https://github.com/ranmaru22/firefox-vertical-tabs/assets/16521734/e6297845-aee0-47c5-bbd3-715552a2db07)

It's the pink-bordered bit. It acts as if there were an `overflow: hidden` or similar on it, but there's not.

Anyway, long story short: somehow this whole behavior is different for different people who use the mod and I have no clue why that is. 🤔

**Comment by klubklubla** [comment]
> This is a really strange one which multiple people on different platforms seem to encounter. The problem is, I could never really figure out what the issue is.
> 
> I thought it's something to do with the sidebar's z-index, but even if I crank that up to some really high number, it works for me. The issue seems to have something to do with what happens to `.urlbarView` when you focus the input. Normally it's set to `display: none`, and then it changes to `block`, but that's not all that's happening. The behavior of this bit of the UI depends on some Firefox internals which user CSS can't touch, namely some custom XML props on `#urlbar`. When you focus it, it gains several custom props like `focus=true`, `open`, and `breakout-extend=true`, and those determine whether the popup part of the URL bar extends beyond its normal box.
> 
> Without it, even with a z-index close to maximum and forced display, it won't show.
> 
> ![image](https://user-images.githubusercontent.com/16521734/280833390-e6297845-aee0-47c5-bbd3-715552a2db07.png)
> 
> It's the pink-bordered bit. It acts as if there were an `overflow: hidden` or similar on it, but there's not.
> 
> Anyway, long story short: somehow this whole behavior is different for different people who use the mod and I have no clue why that is. 🤔
> 
> As a sort-of bandaid solution, you can add a flexible space between the buttons and the URL bar. That pushes it far enough to the right that even in the smallest possible windows it won't overlap with the sidebar.
> 
> ![image](https://user-images.githubusercontent.com/16521734/280834323-09e014e7-db11-4d90-a927-8c4af81fd6cb.png)
> 
> I know that isn't perfect but it at least makes everything visible.

Thank you for your response. I see, it is really weird. When I first set up the side bar with your repository, it worked flawlessly. After some time, presumably after an update, this issue appeared. At that point I played around with the z-index and it solved it completely. But now it happened again and changing the z-indexes did not help anymore.

But thank you for the idea to add space on the left side. I now just moved some stuff from the right to the left side and am completely happy. :)
